server_src/lib/serverCommands.js

```diff
@@ -8,13 +8,22 @@ var wol = require("wake_on_lan");
 class SystemCommands {
   static async rebootOctoFarm() {
     logger.info("Restart OctoFarm server requests");
+    let checkForNamedService = false;
+    // Bit hacky... Make sure OctoFarm has a pid under pm2.... Had to stagger the actual restart command so the return would send to client. - Open to suggestions
     try {
-      const { stdout, stderr } = await exec("pm2 restart OctoFarm");
+      //Attempts to find the named service
+      const { stdout, stderr } = await exec("pm2 pid OctoFarm");
+      if (isNaN(parseInt(stdout))) {
+        throw "No process number returned";
+      }
       logger.info("stdout:", stdout);
-      logger.info("stderr:", stderr);
+      checkForNamedService = true;
+      // If the process had a integer pid succeeded then we are ok to restart, errors are caught by the try, catch
+      exec("pm2 restart OctoFarm");
```

Comment:
Maybe introduce a small timeout as to prevent possibilities of rapid restarts. It's something I got a bit scared off using pm2... however stupid or irrealistic it may sound

Reply:
I did consider that tbh with a setTimout, also could do with diabling the button client side too after a press. Stops any possibility of one then.

Reply:
Ok feel free to do or to skip.

Reply:
I gather the setTimeout would be saved in a variable and checked for its existence here? Our lass is yogaing so can look at that now. I'd rather not people rapid fire, I'll also do the client changes too and make sure that is disabled until we have a response from the server.